### PR TITLE
Add tests to cover OGG Vorbis, and FLAC audio formats

### PR DESCRIPTION
**Pull request by jantonguirao**
#### Why we need this PR?
*Pick one, remove the rest*
- It adds tests for OGG Vorbis and FLAC audio formats

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     *Adjusts tests after adding more audio samples*
     *Add tests for new data formats: OGG Vorbis and FLAC*
 - Affected modules and functionalities:
     *Audio related tests*
 - Key points relevant for the review:
     *Changes in the tests (some existing tests broke after adding more samples, the previous test always tested the same 2 audio files)*
 - Validation and testing:
     *Tests changes only*
 - Documentation (including examples):
     *NA*

**JIRA TASK**: *[DALI-2084]*

### CODE REVIEW

File: dali/test/python/test_operator_nonsilence.py [hunk 1]
```diff
@@ -96,7 +94,7 @@ def test_nonsilence_operator():
     window_sizes = [512, 1024]
     reset_intervals = [-1, 2048, 8192]
     references_power = [None, .0003]
-    cutoff_coeffs = [-10, -20, -30]
+    cutoff_coeffs = [-10, -60, -80]
```
Comment (jantonguirao) [inline dali/test/python/test_operator_nonsilence.py:97]: 20 and 30 failed for some samples. 60 is a more typical "silence" definition. Our implementation is different from librosa, so some differences are expected.

File: dali/test/python/test_torch_pipeline_rnnt.py [hunk 2]
```diff
@@ -15,15 +15,32 @@
...
 import librosa
 import torch
 import math
+import os
 
-dali_extra_path = test_utils.get_dali_extra_path()
+audio_files = test_audio_decoder_utils.get_audio_files('wav')
+npy_files = [os.path.splitext(fpath)[0] + '.npy' for fpath in audio_files]
 
+# From DeepLearningExamples
```
Comment (jantonguirao) [inline dali/test/python/test_torch_pipeline_rnnt.py:29]: original tests relied in npy files that were already normalized.

File: dali/test/python/test_torch_pipeline_rnnt.py [hunk 3]
```diff
@@ -269,25 +282,31 @@ def test_rnnt_data_pipeline():
...
-        assert reference_data[sample_idx].shape[1:] == output_data.shape
-        size = reference_data[sample_idx][1:].flatten().shape[0]
-        assert np.sum(
-                np.isclose(reference_data[sample_idx], output_data, atol=.01, rtol=0)) / size > .99
+        nfeatures = reference_data[sample_idx].shape[1]
+        assert nfeatures == output_data.shape[0]
+        size = nfeatures * audio_len
+        ref = reference_data[sample_idx][:, :, :]
+        out = output_data[:, :]
+        assert np.sum(np.isclose(ref, out, atol=.1, rtol=0)) / size > .9, \
```
Comment (jantonguirao) [inline dali/test/python/test_torch_pipeline_rnnt.py:311]: the expectations were too high when testing on more data. It didn't pass for some samples.

File: dali/test/python/test_audio_decoder_utils.py [hunk 4]
```diff
@@ -24,3 +27,9 @@ def rosa_resample(input, in_rate, out_rate):
     ret[:,c] = a
 
   return ret
+
+def get_audio_files(fmt):
```
Comment (JanuszL) [inline dali/test/python/test_audio_decoder_utils.py:31]: Maybe it could be generalized to `get_files(path, fmt, ext)`
path - "audio"
fmt = ext in this case

File: dali/test/python/test_operator_nonsilence.py (hunk 1 above)
Comment (JanuszL) [reply dali/test/python/test_operator_nonsilence.py:97]: Do you know why it fails? Difference due to compression?

Comment (jantonguirao) [reply dali/test/python/test_operator_nonsilence.py:97]: I have no idea. The non-silent region is significantly different to the librosa based one, for those thresholds. That being said, our implementation calculates energy at a much finer granularity (doesn't use windows, I believe)

File: dali/test/python/test_torch_pipeline_rnnt.py [hunk 5]
```diff
@@ -269,25 +282,31 @@ def test_rnnt_data_pipeline():
...
-    recordings = [np.load(os.path.join(data_path, name)) for name in rec_names]
-    pipe = RnntTrainPipeline(device_id=0, n_devices=1, file_root=data_path,
-                             file_list=os.path.join(data_path, "file_list.txt"),
-                             batch_size=batch_size)
+    recordings = []
+    for fpath in npy_files:
+        arr = np.load(fpath)
+        arr = _convert_samples_to_float32(arr)
+        recordings.append(arr)
+    batch_size = len(recordings)
```
Comment (JanuszL) [inline dali/test/python/test_torch_pipeline_rnnt.py:292]: How many samples it could be?
Maybe we should have a fixed batch size and just use it instead of connecting it with the number of samples?

File: dali/test/python/test_torch_pipeline_rnnt.py [hunk 6]
```diff
@@ -269,25 +282,31 @@ def test_rnnt_data_pipeline():
...
         output_data = dali_out[0].at(sample_idx)
         output_data = np.transpose(output_data, (1, 0))
         audio_len = dali_out[1].at(sample_idx)[0]
         assert audio_len == reference_data[sample_idx].shape[2]
-        assert reference_data[sample_idx].shape[1:] == output_data.shape
-        size = reference_data[sample_idx][1:].flatten().shape[0]
-        assert np.sum(
-                np.isclose(reference_data[sample_idx], output_data, atol=.01, rtol=0)) / size > .99
+        nfeatures = reference_data[sample_idx].shape[1]
+        assert nfeatures == output_data.shape[0]
```
Comment (JanuszL) [inline dali/test/python/test_torch_pipeline_rnnt.py:307]: Is it equivalent to the previous implementation? Does the data has only 2-dimensions?

File: dali/test/python/test_torch_pipeline_rnnt.py (hunk 3 above)
Comment (JanuszL) [reply dali/test/python/test_torch_pipeline_rnnt.py:311]: Does this difference come only from librosa?

File: dali/test/python/test_operator_nonsilence.py (hunk 1 above)
Comment (JanuszL) [reply dali/test/python/test_operator_nonsilence.py:97]: @mzient ?

File: dali/test/python/test_torch_pipeline_rnnt.py (hunk 5 above)
Comment (jantonguirao) [reply dali/test/python/test_torch_pipeline_rnnt.py:292]: it's 43 samples now.

File: dali/test/python/test_torch_pipeline_rnnt.py (hunk 6 above)
Comment (jantonguirao) [reply dali/test/python/test_torch_pipeline_rnnt.py:307]: yes, the first dimension is 1 in the reference (we are calculating the reference sample by sample).

File: dali/test/python/test_torch_pipeline_rnnt.py (hunk 5 above)
Comment (JanuszL) [reply dali/test/python/test_torch_pipeline_rnnt.py:292]: Should be bearable.

**Review by JanuszL — APPROVED** [review]

File: dali/test/python/test_torch_pipeline_rnnt.py (hunk 3 above)
Comment (jantonguirao) [reply dali/test/python/test_torch_pipeline_rnnt.py:311]: I don't think it comes from librosa. This is a whole pipeline, there can be some differences in the implementation of each individual part.  If we wanted, we could spend time analyzing the differences in the pipeline. Something like we have for the SSD pipeline, return the output at every stage and compare it with the reference.

In this PR we haven't changed the DALI implementation, just started testing more data (before it was only 2 samples).

Comment (JanuszL) [reply dali/test/python/test_torch_pipeline_rnnt.py:311]: Just asking.

File: dali/test/python/test_audio_decoder_utils.py (hunk 4 above)
Comment (jantonguirao) [reply dali/test/python/test_audio_decoder_utils.py]: Done

File: dali/test/python/test_operator_nonsilence.py (hunk 1 above)
Comment (jantonguirao) [reply dali/test/python/test_operator_nonsilence.py:96]: I missed this

> It is because we calculate differently the running average

Yes, we calculate the running average differently. Librosa works with windows, while we have sample granularity, I believe.

Comment (JanuszL) [reply dali/test/python/test_operator_nonsilence.py:96]: Understood

**Comment by jantonguirao** [comment]
!build

File: dali/test/python/test_operator_decoders_audio.py [hunk 7]
```diff
@@ -152,3 +152,27 @@ def test_audio_decoder_alias():
...
+  def audio_decoder_pipe(fnames, dtype, downmix=True):
+      encoded, _ = fn.readers.file(files=fnames)
+      decoded, _ = fn.decoders.audio(encoded, dtype=dtype, downmix=downmix)
+      return decoded
+
+  audio_files = get_files(f'db/audio/{fmt}', fmt)
+  npy_files = [os.path.splitext(fpath)[0] + '.npy' for fpath in audio_files]
+  new_pipe = audio_decoder_pipe(audio_files, dtype)
+  npy_pipe = numpy_reader_pipe(npy_files)
+  compare_pipelines(new_pipe, npy_pipe, batch_size, 10, eps=1)
```
Comment (jantonguirao) [inline dali/test/python/test_operator_decoders_audio.py:173]: Note: OGG decoding had a few samples differing by 1. (Comparing our audio decoder and what we used to build the reference, sox). Wav and flac were matching exactly

File: dali/test/python/test_torch_pipeline_rnnt.py [hunk 8]
```diff
@@ -154,17 +171,191 @@ def forward(self, inp, seq_len):
...
+        y = fn.preemphasis_filter(x, preemph_coeff=preemph)
+        return y
+    return dali_run(preemph_pipe(), device=device)
+
+def _testimpl_torch_vs_dali_preemphasis(device):
+    arr = _convert_samples_to_float32(np.load(npy_files[0]))
+    torch_out = torch_preemphasis(arr, 0.97, device=device)
+    dali_out = dali_preemphasis(arr, 0.97, device=device)
+    # DALI and torch differ in the first element:
+    # DALI: y[0] = x[0] - coeff * x[0]
```
Comment (mzient) [inline dali/test/python/test_torch_pipeline_rnnt.py:319]:
```suggestion
    # DALI: y[0] = x[0] - coeff * x[1]
```
?

**Review by mzient — APPROVED** [review]
